modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/consent/TrackingConsentDialog.kt

```diff
@@ -0,0 +1,84 @@
+package au.com.shiftyjelly.pocketcasts.settings.consent
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import au.com.shiftyjelly.pocketcasts.compose.buttons.RowButton
+import au.com.shiftyjelly.pocketcasts.compose.buttons.RowOutlinedButton
+import au.com.shiftyjelly.pocketcasts.compose.components.TextP30
+import au.com.shiftyjelly.pocketcasts.compose.components.TextP50
+import au.com.shiftyjelly.pocketcasts.compose.theme
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun TrackingConsentDialog(
+    onAllow: () -> Unit,
+    onAskAppNotToTrack: () -> Unit,
+) {
+    Dialog(onDismissRequest = { onAskAppNotToTrack() }) {
```

Comment:
We might want to make the dialog modal, an accidental touch outside the dialog will always disallow tracking.

```suggestion
    Dialog(
        onDismissRequest = { onAskAppNotToTrack() },
        properties = DialogProperties(
            dismissOnBackPress = false,
            dismissOnClickOutside = false,
        ),
    ) {
```

Reply:
Thanks for the suggestion. I didn't want them to be forced to make a choice but it's a good question. I will ask.